File: 1518-water-bottles/1518-water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        max_bottle = numBottles
        while numBottles >= numExchange:
            max_bottle += numBottles // numExchange
            numBottles = (numBottles // numExchange) + (numBottles % numExchange)
        return max_bottle
        

File: 1518-water-bottles/1518-water-bottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function(numBottles, numExchange) {
    let max_bottle = numBottles;
    while(numBottles >= numExchange){
        max_bottle += Math.floor(numBottles/numExchange);
        numBottles = Math.floor(numBottles/numExchange) + numBottles%numExchange;
    }
    return max_bottle
    
};